addons/advanced_ballistics/functions/fnc_handleFirePFH.sqf

```diff
@@ -29,11 +28,16 @@
         };
 
         _bullet setVelocity (_bulletVelocity vectorAdd (parseSimpleArray ("ace_advanced_ballistics" callExtension format["simulate:%1:%2:%3:%4:%5:%6", _index, _bulletVelocity, _bulletPosition, wind, ASLToATL(_bulletPosition) select 2, CBA_missionTime toFixed 6])));
+    } else {
+        GVAR(allBullets) set [_forEachIndex, objNull];
+        _deleted = true;
     };
-    nil
-} count +GVAR(allBullets);
+} forEach GVAR(allBullets);
 
-if (GVAR(allBullets) isEqualTo []) then {
-    [_this select 1] call CBA_fnc_removePerFrameHandler;
-    GVAR(BulletPFH) = nil;
+if (_deleted) then {
+    GVAR(allBullets) = GVAR(allBullets) - [objNull];
```

Comment:
Is this quick than `GVAR(allBullets) arrayIntersect GVAR(allBullets)` to remove `nil` (instead of `objNull`) elements?

Alternatively, shouldn't the bullets automatically become `objNull` once they've impacted? Why are we checking `alive` to begin with?

Reply:
the GVAR(allBullets) does not contain bullets but a data array.

Reply:
Ah yeah I see now it's unpacked in the loop and we're just using `objNull` for the removal 👍

Reply:
In that case, what about doing a `select {}` and just running all the code in the select block and ending it with `alive _bullet`?

Reply:
slower. this is currently the fastest method. we had that discussion what the fastest way of that type of filtering an array I think 2x already and we decided and performance checked that this was back than the fastest method. and I just updated those now because I think we missed the 2 back in the day. the same code also runs in the CBA for some PFH systems